src/platform/plugins/shared/discover/public/application/main/state_management/redux/utils.ts

```diff
@@ -29,8 +30,22 @@ type CreateInternalStateAsyncThunk = ReturnType<
   }>
 >;
 
-export const createInternalStateAsyncThunk: CreateInternalStateAsyncThunk =
-  createAsyncThunk.withTypes();
+export const createInternalStateAsyncThunk: CreateInternalStateAsyncThunk = ((
+  ...[typePrefix, payloadCreator, options]: Parameters<CreateInternalStateAsyncThunk>
+) => {
+  return createAsyncThunk(typePrefix, payloadCreator, {
+    ...options,
+    serializeError: (error) => {
+      if (options?.serializeError) {
+        return options.serializeError(error);
+      }
+      if (error instanceof SavedObjectNotFound) {
+        return error;
+      }
+      return miniSerializeError(error);
+    },
```

Comment:
By default Redux async thunks serialize all uncaught errors, but we need the raw `SavedObjectNotFound` instance to handle redirects.